src/Stratis.Bitcoin.Tests/Consensus/ChainedHeaderTreeTest.cs

```diff
@@ -362,11 +362,16 @@ public void ChainHasTwoCheckPoints_ChainCoveringOnlyFirstCheckPointIsPresented_C
 
             // Remaining 5 blocks are presented by peer 1 which do not cover checkpoint 2.
             // InvalidHeaderException should be thrown.
+            List<BlockHeader> violatingHeaders = listOfNewChainHeaders.Skip(5).ToList();
             Action connectAction = () =>
             {
-                cht.ConnectNewHeaders(1, listOfNewChainHeaders.Skip(5).ToList());
+                cht.ConnectNewHeaders(1, violatingHeaders);
             };
 
+            // Make sure headers for violating chain don't exist.
+            foreach (BlockHeader header in violatingHeaders)
```

Comment:
Connect action is not executed yet. Don't you need to connect first for this assert (ie do it after line 375) ?